twitter_api/main.py

```diff
@@ -1,28 +1,278 @@
+import string
+from datetime import datetime
 
+app = Flask(__name__)
+#api = Api(app)
 
 def connect_db(db_name):
     return sqlite3.connect(db_name)
 
-
 @app.before_request
 def before_request():
     g.db = connect_db(app.config['DATABASE'])
+    users = g.db.execute('SELECT access_token FROM auth')
+    
+# implement your views here
+@app.route('/login', methods = ['POST'])
+@json_only
+def login():
+    data = request.get_json()
+    try:
+        post_username = data['username']
+        post_password = data['password']
+    except KeyError:
+        abort(400) # Bad request
+    
+    sql_command = 'SELECT username, password, id FROM user WHERE username = ?'
+    
+    found_user_cursor = g.db.execute(sql_command, [post_username])
+    found_user = found_user_cursor.fetchone()
+    try:
+        username, password, id = found_user
+    except TypeError: # if username did find a match in db
+        abort(404) # Not found
+        
+    post_password = post_password.encode('utf-8') # needs to be in bytes
+    post_password = md5(post_password).hexdigest()
+    
+    if password != post_password:
+        abort(401) # Unauthorized
+        
+    # generate access token
+    char_list = string.ascii_lowercase + string.ascii_uppercase
+    access_token = ''.join(random.choice(char_list) for _ in range(10))
+    
+    sql_command = 'INSERT INTO auth (user_id, access_token) VALUES (?, ?)'
+    g.db.execute(sql_command, [id, access_token])
+    g.db.commit()
+    
+    return_dict = dict(access_token = access_token)
+    response = Response(
+        json.dumps(return_dict), status=201, content_type=JSON_MIME_TYPE)
+    return response
+    
+@app.route('/logout', methods = ['POST'])
+@json_only
+@auth_only
+def logout():
 
+    post_request = request.get_json()
+    post_access_token = post_request['access_token']
 
-# implement your views here
+    # Delete entire row from auth table
+    sql_command = 'DELETE FROM auth WHERE access_token = ?'
+    g.db.execute(sql_command, [post_access_token])
+    g.db.commit()
+    return ('', 204) # No Content
+    
+@app.route('/tweet/<tweet_id>')
+def get_tweet(tweet_id):
 
+    # sql_command = 'SELECT * FROM tweet WHERE id = ?'
+    # tweet_cursor = g.db.execute(sql_command, [tweet_id])
+    # tweet = tweet_cursor.fetchone()
+    # if not tweet:
+    #     abort(404)
+    
+    post_tweet_id, post_user_id, post_created, post_content = get_tweet_info(tweet_id)
+    
+    # if request.method == 'POST': # want to delete tweet
+    #     # need to get token...
+    #     # kinda sucks - rewriting utils.py (why restful-flask is better...
+    #     # we could have separate )
+    #     post_request = request.get_json()
+    #     if request.get_json() is None:
+    #         abort(400) # Bad request
+    #     post_access_token = post_request['access_token']
+    
+    uri = '/tweet/{}'.format(post_tweet_id)
+    
+    # need to get username
+    sql_command = 'SELECT username FROM user WHERE id = ?'
```

Comment:
You can use a JOINed query and get the information for both, the user and the tweet, in a single query.